wallet-next/src/storage.rs

```diff
@@ -34,15 +38,69 @@ VALUES ( ?1 )
     pub async fn read_table(self: &Storage) -> anyhow::Result<String> {
         let recs = sqlx::query!(
             r#"
-SELECT id, value
-FROM penumbra
-ORDER BY id
-LIMIT 1
+            SELECT id, value
+            FROM penumbra
+            ORDER BY id
+            LIMIT 1
         "#
         )
         .fetch_all(&self.pool)
         .await?;
 
         Ok(recs[0].value.clone())
     }
+    /// The last block height we've scanned to, if any.
+    pub async fn last_sync_height(&self) -> anyhow::Result<Option<u64>> {
+        let result = sqlx::query!(
+            r#"
+            SELECT height
+            FROM sync_height
+            ORDER BY height DESC
+            LIMIT 1
+        "#
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        Ok(result[0].value.clone())
+    }
+    pub async fn chain_params(&self) -> anyhow::Result<Vec<u8>> {
+        let result = query!(
+            r#"
+            SELECT bytes
+            FROM chain_params
+            LIMIT 1
+        "#
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        Ok(result[0].value.clone())
+    }
+    pub async fn full_viewing_key(&self) -> anyhow::Result<Vec<u8>> {
+        let result = query!(
+            r#"
+            SELECT bytes
+            FROM full_viewing_key
+            LIMIT 1
+            "#
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        Ok(result)
```

Comment:
Similarly, this should return a `FullViewingKey`, which should be deserializable from proto-encoded bytes in the same way.